src/algorithms/audioproblems/discontinuitydetector.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2006-2016  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#ifndef ESSENTIA_DISCONTINUITYDETECTOR_H
+#define ESSENTIA_DISCONTINUITYDETECTOR_H
+
+#include "algorithm.h"
+#include "algorithmfactory.h"
+#include "essentiamath.h"
+
+namespace essentia {
+namespace standard {
+
+class DiscontinuityDetector : public Algorithm {
+
+  private:
+    Input<std::vector<Real>> _frame;
+    Output<std::vector<Real>> _discontinuityLocations;
+    Output<std::vector<Real>> _discontinuityAmplitues;
+
+    int _order;
+    int _hopSize;
+    int _kernelSize;
+    float _detectionThld;
+    float _energyThld;
+    int _subFrameSize;
+    int _frameSize;
+
+    Algorithm* _medianFilter;
+    Algorithm* _LPC;
+    Algorithm* _windowing;
+
+    // normalize to the max(abs(array))
+    template <typename T> void absNormalize(std::vector<T>& array) {
```

Comment:
This method is useful enough to be placed in essentiamath.h. We already have [normalize](https://github.com/MTG/essentia/blob/master/src/essentia/essentiamath.h#L658) and [normalizeSum](https://github.com/MTG/essentia/blob/master/src/essentia/essentiamath.h#L672) there. Rename ```absNormalize``` to ```normalizeAbs``` for consistency.